From 07ec22258be400fb2d8a8fc0db565bf77c82a5cc Mon Sep 17 00:00:00 2001
From: "Derek J. Clark" <derekjohn.clark@gmail.com>
Date: Mon, 17 Jun 2024 11:49:30 -0700
Subject: [PATCH] Add OrangePi NEO Scancodes

Adds scancodes for the OrangePi NEO Handheld Gaming computer. This
device ships with an AT Translated Set 2 Keyboard device that
provides two buttons, ~~LC (Top Left) and RC (Top Right)~~
Home (front, bottom left) and Gamepad (front, bottom right). The
scancodes do not properly map in Linux. This change maps these
scancodes to ensure the hardware behaves as the OEM expects.

(cherry picked from commit 7c6028bbcbd03f91e1c4b84dcf46b45e9672c2b6)

Related: RHEL-5950
---
 hwdb.d/60-keyboard.hwdb | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/hwdb.d/60-keyboard.hwdb b/hwdb.d/60-keyboard.hwdb
index d271f17005..81ffc69fa1 100644
--- a/hwdb.d/60-keyboard.hwdb
+++ b/hwdb.d/60-keyboard.hwdb
@@ -1657,6 +1657,15 @@ evdev:atkbd:dmi:bvn*:bvr*:bd*:svnOQO*Inc.*:pnOQO*Model*2*:*
  KEYBOARD_KEY_f2=volumedown
  KEYBOARD_KEY_f3=volumeup
 
+###########################################################
+# OrangePi
+###########################################################
+
+# NEO
+evdev:name:AT Translated Set 2 keyboard:dmi:*:svnOrangePi:pnNEO-01:*
+ KEYBOARD_KEY_66=f15                                    # Gamepad (front, bottom right)
+ KEYBOARD_KEY_67=f16                                    # Home (front, bottom left)
+
 ###########################################################
 # Ortek
 ###########################################################
